src/modules/rest/extjs/src/main/java/it/geosolutions/geostore/services/rest/impl/RESTExtJsServiceImpl.java

```diff
@@ -327,9 +352,16 @@ public ExtResourceList getExtResourcesList(
 
         try {
             List<Resource> resources =
-                    filterOutUnadvertisedResources(
+                    handleFoundResources(
```

Comment:
Essentially the same pattern repeated: check availability + set permission flags. Consider combining or reusing code from ResourceServiceImpl to avoid duplication. Possibly move “permission setting” logic to a single place so that the same user-resource checks are consistent across the application.

Reply:
A dedicated `PermissionService` was created to handle resource access checks. However, the specific logic for setting permission flags has been intentionally kept separate within the respective services to ensure no existing functional requirements are inadvertently affected.